### PR TITLE
Draft iterator interface

### PR DESCRIPTION
**Pull request by mschauer**
@cscherrer have a look

### CODE REVIEW

File: src/sfactiter.jl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        end
+    end
+    iterate(FS, ((t => (x, θ)), t_old, (acc, num), Q, b, G, G2))
+end
+
+
+function iterate(FS::FactSampler, (u, t_old, (acc, num), Q, b, G, G2))
+    t, (x, θ) = u
+    n = length(x)
+    t, (x, θ) = u
```
Comment (mschauer) [inline src/sfactiter.jl:52]:
```suggestion
```

File: test/testiter.jl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    c = .7*[norm(Γ[:, i], 2) for i in 1:d]
+
+    Z = ZigZag(0.9Γ, x0*0)
+    T = 1000.0
+
+    sampler = FactSampler(∇ϕ, t0 =>(x0, θ0), c, Z, Γ)
+
+    tr = trace(sampler, T)
+    dt = 0.5
+    @test mean(trace) < 2/sqrt(T)
```
Comment (mschauer) [inline test/testiter.jl:26]:
```suggestion
    @test mean(tr) < 2/sqrt(T)
```

**Comment by cscherrer** [comment]
@mschauer did this just not work out?

**Comment by mschauer** [comment]
All is well, sorry, it ended up as part of the other PR #71

**Pull request closed without merging**